CommonTools/PileupAlgos/plugins/PuppiPhoton.h

```diff
@@ -37,11 +37,12 @@ class PuppiPhoton : public edm::stream::EDProducer<> {
 	edm::EDGetTokenT< edm::ValueMap<float> > tokenWeights_; 
 	edm::EDGetTokenT< edm::ValueMap<bool>  > tokenPhotonId_; 
 	double pt_;
+	double eta_;
 	bool   usePFRef_;
 	bool   usePhotonId_;
 	std::vector<double>  dRMatch_;
 	std::vector<int32_t> pdgIds_; 
-        std::unique_ptr< PFOutputCollection > corrCandidates_;
+        std::auto_ptr< PFOutputCollection > corrCandidates_;
```

Comment:
auto_ptr is deprecated. Support for it will soon be removed. Revert your changes and go back to unique_ptr, here and on lines 100, 159, 164, and 168 of  PuppiPhoton.cc.